torch/_inductor/codegen/wrapper.py

```diff
@@ -27,18 +28,19 @@ def buffer_reuse_key(node: ir.Buffer):
     )
 
 
-def make_buffer_reuse(old, new):
+def make_buffer_reuse(old, new, del_needed, declare, ending, as_strided):
     assert old.get_dtype() == new.get_dtype()
     del_line = ""
-    if old.get_name() not in V.graph.get_output_names():
-        del_line = f"; del {old.get_name()}"
+    if del_needed:
```

Comment:
Even in C++ we should clear the reference to the old tensor for peak memory usage.

Reply:
Added `tensor.reset()` here for cpp wrapper.